### DISCUSSION TITLE
proposal: Remove args support for databend

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#8303] [Ideas]
Databend has so many configs that it's nearly impossible for users to config them by args.

How about removing them all (except `--config`)?

After this change, users can only config by databend by config files and env.

**Comment by everpcpc** [comment]
Some arguments like listen ip address can only be determined at runtime and vary in instance, they need to be able to set by argument.

**Reply by Xuanwo** [reply]
How about removing part of args support like `storage`?

**Comment by ZhiHanZ** [comment]
I think it is an non-trivial breaking change, as we could see that many of our CI scripts are using arguments, maybe we should take several weeks for migration, besides in databend-meta there do not have much configurations in config file, we could rely on arguments for much readable configurations.
